src/RF24Logger.h

```diff
@@ -0,0 +1,167 @@
+ * library.  If this is what you want to do, use the GNU Lesser General
+ * Public License instead of this License.
+ */
+
+#ifndef SRC_RF24LOGGER_H_
+#define SRC_RF24LOGGER_H_
+
+#include <WString.h>
+#include "RF24LogLevel.h"
+#include "RF24LogHandler.h"
+
+/**
+ * This is the end-user's access point into the world of logging messages.
+ */
+class RF24Logger
+{
+private:
+   RF24LogHandler *handler;
+
+public:
+
+   /**
+    * Initializes the appender to nullptr
+    */
+   RF24Logger();
+
+   /**
+    * set the instance's handler
+    * @param handler The log handler where the messages will be redirected.
+    */
+   void setHandler(RF24LogHandler *handler);
+
+   /**
+    * ouput an ERROR message
+    * @param vendorId A scoping identity of the message's origin
+    * @param message The message to output
+    * @param args consumable arguments
+    */
+   template<class T> void error(
+         const __FlashStringHelper *vendorId, T message, ...)
+   {
+      if (handler == nullptr)
+      {
+         return;
+      }
+      va_list args;
+      va_start(args, message);
+      handler->log(RF24LogLevel::ERROR, vendorId, message, &args);
+   }
+
+   /**
+    * output a message to WARN the reader
+    * @param vendorId A scoping identity of the message's origin
+    * @param message The message to output
+    * @param args consumable arguments
+    */
+   template<class T> void warn(
+         const __FlashStringHelper *vendorId, T message, ...)
+   {
+      if (handler == nullptr)
+      {
+         return;
+      }
+      va_list args;
+      va_start(args, message);
+      handler->log(RF24LogLevel::WARN, vendorId, message, &args);
+   }
+
+   /**
+    * output an INFO message
+    * @param vendorId A scoping identity of the message's origin
+    * @param message The message to output
+    * @param args consumable arguments
+    */
+   template<class T> void info(
+         const __FlashStringHelper *vendorId, T message, ...)
+   {
+      if (handler == nullptr)
+      {
+         return;
+      }
+      va_list args;
+      va_start(args, message);
+      handler->log(RF24LogLevel::INFO, vendorId, message, &args);
+   }
+
+   /**
+    * output a message to help developers DEBUG their source code
+    * @param vendorId A scoping identity of the message's origin
+    * @param message The message to output
+    * @param args consumable arguments
+    */
+   template<class T> void debug(
+         const __FlashStringHelper *vendorId, T message, ...)
+   {
+      if (handler == nullptr)
+      {
+         return;
+      }
+      va_list args;
+      va_start(args, message);
+      handler->log(RF24LogLevel::DEBUG, vendorId, message, &args);
+   }
+
+   /**
+    * output a TRACE message that points to an event's instigator
+    * @param vendorId A scoping identity of the message's origin
+    * @param message The message to output
+    * @param args consumable arguments
+    */
+   template<class T> void trace(

```

Comment:
I think we can just remove trace. There is good enough verbosity with sublevels.

Reply:
agreed

Reply:
Already done.